packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1000,7 +1000,7 @@ class TextSelectionGestureDetectorBuilder {
   /// Handler for [TextSelectionGestureDetector.onSingleLongTapStart].
   ///
   /// By default, it selects text position specified in [details] if selection
-  /// is enabled.
+  /// is enabled. If widget is not focused, it selects the word that is tapped on.
```

Comment:
This seems to be iOS specific. On Android long-pressing always selects the word from what I'm seeing.